### ISSUE TITLE
User story - Job post deletion

### ISSUE THREAD

**Issue by stephendawsondev** [stephendawsondev/StackPortfolio#31]
### If applicable, add a related Epic.

Epic: #28 

### User Story

As a **recruiter user**, I can **delete a job post** so that I can **remove out-of-date posts**.

### Acceptance Criteria

- On job post page or job post management page, the recruiter should have a button to remove their own job post
- They should get a warning before deleting their job post
- When the post is deleted, it should also be deleted from the database

### Please ensure you've completed these tasks before marking the issue as done.

- [x] Reviewed the user story for clarity and completeness.
- [x] Identified and linked related issues or epics.
- [ ] Created or updated the necessary documentation.
- [ ] Written unit and/or integration tests for functionality.
- [x] All code changes have been manually tested to ensure they meet the acceptance criteria.
- [x] Updated the issue with any relevant notes or feedback (to date).
- [x] Merged into the main branch and deployed to production (if applicable).

### Please add any further tasks that need to be completed before the issue can be marked as done.

- [ ] Set up the delete post path with `<post-id>/delete`
- [x] Create delete button
- [x] Create warning before deletion
